processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -98,287 +167,248 @@ public void start()
   public void stop()
   {
     startStopLock.writeLock().lock();
-    try {
-      if (started.getAndSet(false)) {
-        if (lookupSnapshotTaker != null) {
-          lookupSnapshotTaker.takeSnapshot(getAllAsList());
+
+    if (started) {
+      try {
+        LOG.info("LookupReferencesManager is stopping.");
+        started = false;
+
+        if (exec != null) {
+          exec.shutdownNow();
+          exec = null;
         }
-        LOGGER.info("Stopping lookup factory references manager");
-        for (String lookupName : lookupMap.keySet()) {
-          lookupMap.remove(lookupName).close();
+
+        for (Map.Entry<String, LookupExtractorFactoryContainer> e : lookupMap.entrySet()) {
+          try {
+            LOG.info("Closing lookup [%s]", e.getKey());
+            if (!e.getValue().getLookupExtractorFactory().close()) {
+              LOG.error("Failed to close lookup [%s].");
+            }
+          }
+          catch (Exception ex) {
+            LOG.error(ex, "Failed to close lookup [%s].", e.getKey());
+          }
         }
+
+        lookupMap.clear();
       }
-    }
-    finally {
+      finally {
+        startStopLock.writeLock().unlock();
+      }
+      LOG.info("LookupReferencesManager is stopped.");
+    } else {
       startStopLock.writeLock().unlock();
     }
   }
 
-  /**
-   * @param lookupName             name of the lookupExtractorFactory object
-   * @param lookupExtractorFactory {@link LookupExtractorFactory} implementation reference.
-   *
-   * @return true if the lookup is added otherwise false.
-   *
-   * @throws IllegalStateException If the manager is closed or if start of lookup returns false.
-   */
-  public boolean put(String lookupName, final LookupExtractorFactory lookupExtractorFactory)
+  public void add(String lookupName, LookupExtractorFactoryContainer lookupExtractorFactoryContainer)
   {
+    assertStarted();
+
     try {
-      startStopLock.readLock().lockInterruptibly();
-    }
-    catch (InterruptedException e) {
-      throw Throwables.propagate(e);
-    }
-    try {
-      assertStarted();
-      if (lookupMap.containsKey(lookupName)) {
-        LOGGER.warn("lookup [%s] is not add, another lookup with the same name already exist", lookupName);
-        return false;
+      if (!queue.offer(new LoadNotice(lookupName, lookupExtractorFactoryContainer), 1, TimeUnit.MILLISECONDS)) {
```

Comment:
well that just means ... "do not wait if queue is full" ... GC or anything else does not have any impact on that... wait comes into picture only if the queue was full and we are essentially saying not to wait at all.
so it is alright and no need to be configurable..... coordinator will resubmit the request if add to queue wasn't successful.